skycoin/skywire
# Public Autoconnect [API | Summary]

Did you run `make format && make check`? Yes

Fixes #904 

 Changes:	
- Add `public_autoconnect` field to `/visors/{pk}/summary` endpoint
- Make new endpoint `PUT /visors/{pk}/public-autoconnect` to change value of `public_autoconnect`. The body of request should be same as `{"public_autoconnect":true}`.

How to test this PR:
- Check summary on network monitor to see `public_autoconnect` field.
- Make `PUT` request on `/visors/{pk}/public-autoconnect` endpoint with `{"public_autoconnect":true}` or `{"public_autoconnect":false}` to change its value.
- Check summary again for changes.

Note:
As I talked to @jdknives , after this changes we need restart/reboot visor to start [connector routine](https://github.com/skycoin/skywire/blob/63b29ae4625d8407acef34208c24efd7e763596c/pkg/visor/init.go#L667). Maybe after successful change, we should show some alert/notification to user for reboot visor. @Senyoret1
